File: src/B_5543/B_5543.java
package B_5543;

import java.io.*;
import java.util.*;

public class B_5543 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int[] bur = new int[3];
        for (int i = 0; i < 3; ++i) {
            bur[i] = Integer.parseInt(br.readLine());
        }
        int mBur = Arrays.stream(bur).min().getAsInt();

        int[] sid = new int[2];
        for (int i = 0; i < 2; ++i) {
            sid[i] = Integer.parseInt(br.readLine());
        }

        int mSid = Arrays.stream(sid).min().getAsInt();

        System.out.println((mBur+mSid)-50);
    }
}

File: src/B_5543/B_5543.kt
package B_5543

import java.io.*
import java.util.*

fun main() = with(BufferedReader(InputStreamReader(System.`in`))){
    val bur: IntArray = IntArray(3)
    for(i in 0 until bur.count()){
        bur[i] = readLine().toInt()
    }
    val mBur: Int = Arrays.stream(bur).min().asInt

    val sid: IntArray = IntArray(2)
    for(i in 0 until sid.count()){
        sid[i] = readLine().toInt()
    }
    val mSid: Int = Arrays.stream(sid).min().asInt
    print((mBur+mSid)-50)
}